src/sentry/static/sentry/app/views/eventsV2/eventView.tsx

```diff
@@ -287,59 +403,69 @@ class EventView {
       project: saved.projects,
       start: saved.start,
       end: saved.end,
-      range: saved.range,
       sorts: fromSorts(saved.orderby),
       tags: [],
+      statsPeriod: saved.range,
+      environment: collectQueryStringByKey(
+        {
+          environment: (saved as SavedQuery).environment as string[],
+        },
+        'environment'
+      ),
     });
   }
 
   toNewQuery(): NewQuery {
-    const orderby = this.sorts ? encodeSorts(this.sorts)[0] : undefined;
+    const orderby = this.sorts.length > 0 ? encodeSorts(this.sorts)[0] : undefined;
+
     return {
-      id: this.id,
       version: 2,
+      id: this.id,
       name: this.name || '',
+      fields: this.getFields(),
+      fieldnames: this.getFieldNames(),
+      orderby,
+      // TODO: tags?
       query: this.query || '',
```

Comment:
@markstory when `query` is a blank string; there's an error from the backend requiring this field to be non-empty. This error occurs when a user attempts to save the "all events" query as a new saved query.

Reply:
We could omit this key if the query is emptyish.

Reply:
Looks like it works when the `query` key is omitted. I've addressed this in https://github.com/getsentry/sentry/pull/15045/commits/97a0616b52e50531343fcf569817bf1fc3a09f49